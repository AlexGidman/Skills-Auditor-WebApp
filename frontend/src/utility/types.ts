export const SYSTEM_ROLES = ["StaffUser", "Manager", "Admin"];
export const STAFF_USER = SYSTEM_ROLES[0];
export const MANAGER_SR = SYSTEM_ROLES[1];
export const ADMIN = SYSTEM_ROLES[2];

export const JOB_ROLES = ["Mid-Level Developer", "Senior Developer", "Manager"];
export const MIDLEVEL_DEVELOPER = JOB_ROLES[0];
export const SENIOR_DEVELOPER = JOB_ROLES[1];
export const MANAGER_JR = JOB_ROLES[2];

export const TOKEN_COOKIE = "token";

// Week in seconds
export const TOKEN_LIFE = 60 * 60 * 24 * 7;

export const SKILL_LEVELS = ["None", "Basic", "Intermediate", "High", "Expert"];
export const NONE = SKILL_LEVELS[0];
export const BASIC = SKILL_LEVELS[1];
export const INTERMEDIATE = SKILL_LEVELS[2];
export const HIGH = SKILL_LEVELS[3];
export const EXPERT = SKILL_LEVELS[4];

export type User = {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
};
