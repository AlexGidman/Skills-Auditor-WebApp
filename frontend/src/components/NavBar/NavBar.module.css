@import "../../index.css";

.component {
    color: var(--c-secondary);
    background-color: var(--c-primary);
    width: calc(var(--s1) * 15);
    height: 100%;
    padding: var(--s1);
    position: fixed;
    z-index: 100;
    top: 0;
    left: 0;
    overflow-x: hidden;
    transition: 0.5s;
}

.button {
    background-color: inherit;
    border: none;
    cursor: pointer;
}
.button:hover,
.button:active,
.active {
    background-color: var(--c-secondary);
}

.hamburgerLinks {
    background-color: var(--c-secondary);
    width: 100%;
    height: 100%;
    position: fixed;
    margin-left: calc(-1 * var(--s1));
    margin-top: var(--s1);
    overflow: scroll;
}

.visible {
    display: block;
}

.hidden {
    display: none;
}

.link {
    font-size: large;
}

.listItem {
    margin: var(--s1) 0;
    text-align: center;
}

.logo {
    composes: icon from "../../index.css";
    margin-left: var(--s5);
}

.hamburgerIcon {
    composes: icon from "../../index.css";
}

.signoutButton {
    composes: link;
    width: 12.5rem;
}
