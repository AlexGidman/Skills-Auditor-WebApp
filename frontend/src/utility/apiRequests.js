import axios from "axios";

const categoryEndpoint = "/api/category";
const skillEndpoint = "/api/skill";
const userEndpoint = "/api/user";
const directReportEndpoint = "/api/directreport";
const loginEndpoint = "/api/login";
const userSkillEndpoint = "/api/userskill";

//CATEGORY
export const getAllCategories = async (authHeaders) => {
    return axios.get(categoryEndpoint, authHeaders);
};

export const getCategory = async (authHeaders, id) => {
    return axios.get(`${categoryEndpoint}/${id}`, authHeaders);
};

export const addCategory = async (authHeaders, name) => {
    const data = {
        name,
    };
    return axios.post(`${categoryEndpoint}`, data, authHeaders);
};

export const updateCategory = async (authHeaders, id, name) => {
    const data = {
        id,
        name,
    };
    return axios.put(categoryEndpoint, data, authHeaders);
};

export const deleteCategory = async (authHeaders, id) => {
    const data = { id };
    return axios.delete(categoryEndpoint, { ...authHeaders, data });
};

//SKILL
export const getAllSkills = async (authHeaders) => {
    return axios.get(skillEndpoint, authHeaders);
};

export const getSkill = async (authHeaders, id) => {
    return axios.get(`${skillEndpoint}/${id}`, authHeaders);
};

export const addSkill = async (authHeaders, name, category_id) => {
    const data = {
        name,
        category_id,
    };
    return axios.post(`${skillEndpoint}`, data, authHeaders);
};

export const updateSkill = async (authHeaders, id, name, categoryId) => {
    const data = {
        id,
        name,
        category_id: categoryId,
    };
    return axios.put(skillEndpoint, data, authHeaders);
};

export const deleteSkill = async (authHeaders, id) => {
    const data = { id };
    return axios.delete(skillEndpoint, { ...authHeaders, data });
};

//USER
export const getUserDetails = async (authHeaders, id) => {
    return axios.get(`${userEndpoint}/${id}`, authHeaders);
};

export const getUserFromToken = async (authHeaders) => {
    return axios.get(`${userEndpoint}/token`, authHeaders);
};

export const updateUserDetails = async (
    authHeaders,
    id,
    forename,
    surname,
    email,
    password,
    job_role,
    system_role,
) => {
    const data = {
        id,
        forename,
        surname,
        email,
        password,
        job_role,
        system_role,
    };

    return axios.put(`${userEndpoint}`, data, authHeaders);
};

export const addUser = async (
    authHeaders,
    forename,
    surname,
    email,
    password,
    job_role,
    system_role,
) => {
    const data = {
        forename,
        surname,
        email,
        password,
        job_role,
        system_role,
    };
    return axios.post(userEndpoint, data, authHeaders);
};

export const deleteUser = async (authHeaders, id) => {
    const data = { id };
    return axios.delete(userEndpoint, { ...authHeaders, data });
};

//DIRECT REPORTS
export const getAllDirectReports = async (authHeaders, id) => {
    return axios.get(`${directReportEndpoint}/${id}`, authHeaders);
};

export const addDirectReport = async (authHeaders, userId, reportId) => {
    const data = {
        user_id: userId,
        report_id: reportId,
    };
    return axios.post(directReportEndpoint, data, authHeaders);
};

//STAFF SKILLS
export const getAllStaffSkills = async (authHeaders, id) => {
    return axios.get(`${userSkillEndpoint}/byuser/${id}`, authHeaders);
};

export const getStaffSkill = async (authHeaders, id) => {
    return axios.get(`${userSkillEndpoint}/${id}`, authHeaders);
};

export const deleteStaffSkill = async (authHeaders, id) => {
    const data = { id };
    return axios.delete(userSkillEndpoint, { ...authHeaders, data });
};

export const addUserSkill = async (
    authHeaders,
    user_id,
    skill_id,
    skill_level,
    notes,
    expiry_date,
) => {
    const data = {
        user_id,
        skill_id,
        skill_level,
        notes,
        expiry_date,
    };
    return axios.post(`${userSkillEndpoint}`, data, authHeaders);
};

export const updateStaffSkill = async (authHeaders, id, skill_level, notes) => {
    const data = {
        id,
        skill_level,
        notes,
    };

    return axios.put(`${userSkillEndpoint}`, data, authHeaders);
};

//LOGIN

export const postLogin = async (email, password) => {
    const credentials = { email, password };
    return axios.post(`${loginEndpoint}`, credentials);
};

export const validateToken = async (authHeaders) => {
    return axios.get(`${loginEndpoint}/validate`, authHeaders);
};
