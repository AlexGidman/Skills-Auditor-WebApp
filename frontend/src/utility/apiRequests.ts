import axios, { AxiosRequestHeaders } from "axios";

const categoryEndpoint = "/api/category";
const skillEndpoint = "/api/skill";
const userEndpoint = "/api/user";
const directReportEndpoint = "/api/directreport";
const loginEndpoint = "/api/login";
const userSkillEndpoint = "/api/userskill";

//CATEGORY
export const getAllCategories = async (authHeaders: AxiosRequestHeaders) => {
    return axios.get(categoryEndpoint, authHeaders);
};

export const getCategory = async (authHeaders: AxiosRequestHeaders, id: string) => {
    return axios.get(`${categoryEndpoint}/${id}`, authHeaders);
};

export const addCategory = async (authHeaders: AxiosRequestHeaders, name: string) => {
    const data = {
        name,
    };
    return axios.post(`${categoryEndpoint}`, data, authHeaders);
};

export const updateCategory = async (
    authHeaders: AxiosRequestHeaders,
    id: string,
    name: string,
) => {
    const data = {
        id,
        name,
    };
    return axios.put(categoryEndpoint, data, authHeaders);
};

export const deleteCategory = async (authHeaders: AxiosRequestHeaders, id: string) => {
    const data = { id };
    return axios.delete(categoryEndpoint, { ...authHeaders, data });
};

//SKILL
export const getAllSkills = async (authHeaders: AxiosRequestHeaders) => {
    return axios.get(skillEndpoint, authHeaders);
};

export const getSkill = async (authHeaders: AxiosRequestHeaders, id: string) => {
    return axios.get(`${skillEndpoint}/${id}`, authHeaders);
};

export const addSkill = async (
    authHeaders: AxiosRequestHeaders,
    name: string,
    categoryId: string,
) => {
    const data = {
        name,
        categoryId,
    };
    return axios.post(`${skillEndpoint}`, data, authHeaders);
};

export const updateSkill = async (
    authHeaders: AxiosRequestHeaders,
    id: string,
    name: string,
    categoryId: string,
) => {
    const data = {
        id,
        name,
        categoryId: categoryId,
    };
    return axios.put(skillEndpoint, data, authHeaders);
};

export const deleteSkill = async (authHeaders: AxiosRequestHeaders, id: string) => {
    const data = { id };
    return axios.delete(skillEndpoint, { ...authHeaders, data });
};

//USER
export const getUserDetails = async (authHeaders: AxiosRequestHeaders, id: string) => {
    return axios.get(`${userEndpoint}/${id}`, authHeaders);
};

export const getUserFromToken = async (authHeaders: AxiosRequestHeaders) => {
    return axios.get(`${userEndpoint}/token`, authHeaders);
};

export const updateUserDetails = async (
    authHeaders: AxiosRequestHeaders,
    id: string,
    firstName: string,
    lastName: string,
    email: string,
    password: string,
    jobRole: string, // TODO: change when job role type changes
    systemRole: string, // TODO: change when job system type changes
) => {
    const data = {
        id,
        firstName,
        lastName,
        email,
        password,
        jobRole,
        systemRole,
    };

    return axios.put(`${userEndpoint}`, data, authHeaders);
};

export const addUser = async (
    authHeaders: AxiosRequestHeaders,
    firstName: string,
    lastName: string,
    email: string,
    password: string,
    jobRole: string, // TODO: change when job role type changes
    systemRole: string, // TODO: change when job system type changes
) => {
    const data = {
        firstName,
        lastName,
        email,
        password,
        jobRole,
        systemRole,
    };
    return axios.post(userEndpoint, data, authHeaders);
};

export const deleteUser = async (authHeaders: AxiosRequestHeaders, id: string) => {
    const data = { id };
    return axios.delete(userEndpoint, { ...authHeaders, data });
};

//DIRECT REPORTS
export const getAllDirectReports = async (authHeaders: AxiosRequestHeaders, id: string) => {
    return axios.get(`${directReportEndpoint}/${id}`, authHeaders);
};

export const addDirectReport = async (
    authHeaders: AxiosRequestHeaders,
    userId: string,
    reportId: string,
) => {
    const data = {
        userId: userId,
        reportId: reportId,
    };
    return axios.post(directReportEndpoint, data, authHeaders);
};

//STAFF SKILLS
export const getAllStaffSkills = async (authHeaders: AxiosRequestHeaders, id: string) => {
    return axios.get(`${userSkillEndpoint}/byuser/${id}`, authHeaders);
};

export const getStaffSkill = async (authHeaders: AxiosRequestHeaders, id: string) => {
    return axios.get(`${userSkillEndpoint}/${id}`, authHeaders);
};

export const deleteStaffSkill = async (authHeaders: AxiosRequestHeaders, id: string) => {
    const data = { id };
    return axios.delete(userSkillEndpoint, { ...authHeaders, data });
};

export const addUserSkill = async (
    authHeaders: AxiosRequestHeaders,
    userId: string,
    skillId: string,
    skillLevel: number, // TODO: chnage this when skill level type changes
    notes: string,
    expiryDate: Date,
) => {
    const data = {
        userId,
        skillId,
        skillLevel,
        notes,
        expiryDate,
    };
    return axios.post(`${userSkillEndpoint}`, data, authHeaders);
};

export const updateStaffSkill = async (
    authHeaders: AxiosRequestHeaders,
    id: string,
    skillLevel: number, // TODO: chnage this when skill level type changes
    notes: string,
) => {
    const data = {
        id,
        skillLevel,
        notes,
    };

    return axios.put(`${userSkillEndpoint}`, data, authHeaders);
};

//LOGIN

export const postLogin = async (email: string, password: string) => {
    const credentials = { email, password };
    return axios.post(`${loginEndpoint}`, credentials);
};

export const validateToken = async (authHeaders: AxiosRequestHeaders) => {
    return axios.get(`${loginEndpoint}/validate`, authHeaders);
};
