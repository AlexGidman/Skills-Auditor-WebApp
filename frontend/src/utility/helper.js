import { useEffect, useState } from "react";
import { useOutletContext, useNavigate } from "react-router-dom";
import {
    ADMIN,
    BASIC,
    EXPERT,
    HIGH,
    NONE,
    INTERMEDIATE,
    MANAGER_SR,
    SYSTEM_ROLES,
    TOKEN_COOKIE,
} from "./types";
import { useCookies } from "react-cookie";

export const useLogout = () => {
    const [, , removeCookie] = useCookies(TOKEN_COOKIE);
    const navigate = useNavigate();
    const [logout, setLogout] = useState(false);

    useEffect(() => {
        if (logout) {
            setLogout(false);
            removeCookie(TOKEN_COOKIE);
            navigate("/login");
        }
    }, [logout, navigate, removeCookie]);

    return () => setLogout(true);
};

export const hyphenate = (string) => {
    return string.trim().replace(" ", "-");
};

export const useAPI = (apiCallback, varArray = [], lazy = false, auth = true) => {
    const [cookies] = useCookies(TOKEN_COOKIE);
    const navigate = useNavigate();

    const [callAPI, setCallAPI] = useState(!lazy);
    const [data, setData] = useState(null);
    const [loading, setLoading] = useState(!lazy);
    const [error, setError] = useState(null);

    const callback = () => {
        setLoading(true);
        setCallAPI(true);
    };

    const authHeaders = {
        headers: {
            authorization: cookies[TOKEN_COOKIE],
        },
    };

    useEffect(() => {
        if (callAPI) {
            apiCallback
                .apply(null, auth ? [authHeaders, ...varArray] : [...varArray])
                .then((response) => {
                    setData(response.data);
                })
                .catch((error) => {
                    if (unauthorisedError(error)) {
                        navigate("/login");
                    } else {
                        setError(error.response.data.error);
                    }
                })
                .finally(() => {
                    setLoading(false);
                    setCallAPI(false);
                });
        }
    }, [apiCallback, auth, callAPI, navigate, varArray]);

    return { data, loading, error, callback };
};

export const getSelectOptionsFromArray = (valArray = [], labelArray = []) => {
    return valArray.length === labelArray.length
        ? valArray.map((val, i) => {
              return { value: val, label: labelArray[i] };
          })
        : valArray.map((val) => {
              return { value: val, label: val };
          });
};

export const getAssignableSystemRoleOptions = (currentUserSystemRole) => {
    const systemRolesOptions = getSelectOptionsFromArray(SYSTEM_ROLES);
    switch (currentUserSystemRole) {
        case ADMIN:
            return systemRolesOptions;
        case MANAGER_SR:
            return systemRolesOptions.filter((option) => {
                return option.label !== ADMIN && option;
            });
        default:
            return systemRolesOptions.filter((option) => {
                return option.label !== ADMIN && option.label !== MANAGER_SR && option;
            });
    }
};

export const isValidString = (val, min = 0, max = 100) => {
    return !!val && val.length >= min && val.length <= max;
};

export const isValidEmail = (val) => {
    return /\S+@\S+\.\S+/.test(val);
};

export const isValidUser = (user) => {
    return (
        !!user.firstName &&
        isValidString(user.firstName, 1) &&
        !!user.lastName &&
        isValidString(user.lastName, 1) &&
        !!user.email &&
        isValidEmail(user.email) &&
        (!!user.password ? isValidString(user.password, 10, 16) : true)
    );
};

export const useIsAdminOrManager = () => {
    const [currentUser, setShowToast] = useOutletContext();
    const navigate = useNavigate();

    useEffect(() => {
        if (
            currentUser &&
            currentUser.system_role !== ADMIN &&
            currentUser.system_role !== MANAGER_SR
        ) {
            setShowToast({ error: "Unauthorised route" });
            navigate("/");
        }
    });
};

const unauthorisedError = (error) => error.response.status === 401 || error.response.status === 403;

export const formatDate = (_date) => {
    if (!_date) return "N/A";
    let newDate = new Date(_date);
    newDate.toLocaleDateString();
    return newDate.toLocaleDateString();
};

export const formatSkillLevel = (_skillLevel) => {
    if (_skillLevel === 5) {
        return EXPERT;
    } else if (_skillLevel === 4) {
        return HIGH;
    } else if (_skillLevel === 3) {
        return INTERMEDIATE;
    } else if (_skillLevel === 2) {
        return BASIC;
    } else {
        return NONE;
    }
};

export const convertSkillLevelName = (_skillLevel) => {
    if (_skillLevel === "Expert") {
        return 5;
    } else if (_skillLevel === "High") {
        return 4;
    } else if (_skillLevel === "Intermediate") {
        return 3;
    } else if (_skillLevel === "Basic") {
        return 2;
    } else {
        return 1;
    }
};

export const expiryDateNullFormatter = (_expiryDate) => {
    if (_expiryDate === null) {
        return "N/A";
    } else {
        return _expiryDate;
    }
};
