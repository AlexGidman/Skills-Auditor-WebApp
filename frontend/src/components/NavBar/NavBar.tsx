import React, { useState } from "react";
import { LinkButton } from "../LinkButton/LinkButton";
import { Button } from "../Button/Button";

import { STAFF_USER, MANAGER_SR, ADMIN } from "../../utility/types";
import { Hamburger, Logo } from "../../icons";

import styles from "./NavBar.module.css";
import cx from "classnames";
import { hyphenate, useLogout } from "../../utility/helper";

const navLinks = [
    {
        name: "Home",
        path: "/",
        auth: [STAFF_USER, MANAGER_SR, ADMIN],
    },
    {
        name: "My Details",
        path: "/mydetails",
        auth: [STAFF_USER, MANAGER_SR, ADMIN],
    },
    {
        name: "My Skills",
        path: "/myskills",
        auth: [STAFF_USER, MANAGER_SR, ADMIN],
    },
    {
        name: "View Staff",
        path: "/viewstaff",
        auth: [MANAGER_SR, ADMIN],
    },
    {
        name: "Add Staff",
        path: "/addstaff",
        auth: [MANAGER_SR, ADMIN],
    },
    {
        name: "View Skills",
        path: "/viewskills",
        auth: [MANAGER_SR, ADMIN],
    },
    {
        name: "Add Skill",
        path: "/addskill",
        auth: [MANAGER_SR, ADMIN],
    },
    {
        name: "View Categories",
        path: "/viewcategories",
        auth: [MANAGER_SR, ADMIN],
    },
];

interface Props extends React.ComponentProps<"nav"> {
    systemRole: string; // TODO: change this type when system role changes
}

export const NavBar = ({ systemRole, className, ...rest }: Props): React.ReactElement => {
    const processLogout = useLogout();

    return (
        <nav className={cx(styles.component, className)} {...rest}>
            <Logo className={styles.logo} aria-hidden />
            <ul>
                {navLinks.map((link, index) => {
                    if (link.auth.includes(systemRole)) {
                        return (
                            <li
                                className={styles.listItem}
                                key={`nav-${hyphenate(link.name)}-${index}`}
                            >
                                <LinkButton
                                    path={link.path}
                                    name={link.name}
                                    className={styles.link}
                                />
                            </li>
                        );
                    }
                    return null;
                })}
                <li className={styles.listItem}>
                    <Button
                        className={styles.signoutButton}
                        name="Sign Out"
                        onClick={() => processLogout()}
                    >
                        Sign out
                    </Button>
                </li>
            </ul>
        </nav>
    );
};

export const HamburgerNavBar = ({ systemRole, className, ...rest }: Props): React.ReactElement => {
    const [menuVisible, setMenuVisible] = useState(false);
    const processLogout = useLogout();

    return (
        <nav className={className} {...rest}>
            <button
                className={cx(styles.button, menuVisible && styles.active)}
                onClick={() => setMenuVisible(!menuVisible)}
                aria-label={"Navigation"}
            >
                <Hamburger className={styles.hamburgerIcon} aria-hidden />
            </button>
            <ul className={cx(styles.hamburgerLinks, menuVisible ? styles.visible : styles.hidden)}>
                {navLinks.map((link, index) => {
                    if (link.auth.includes(systemRole)) {
                        return (
                            <li
                                className={styles.listItem}
                                key={`hamburger-${hyphenate(link.name)}-${index}`}
                            >
                                <LinkButton
                                    path={link.path}
                                    name={link.name}
                                    onClick={() => setMenuVisible(false)}
                                />
                            </li>
                        );
                    }
                    return null;
                })}
                <li className={styles.listItem}>
                    <Button
                        className={styles.signoutButton}
                        name="Sign Out"
                        onClick={() => processLogout()}
                    >
                        Sign out
                    </Button>
                </li>
            </ul>
        </nav>
    );
};
