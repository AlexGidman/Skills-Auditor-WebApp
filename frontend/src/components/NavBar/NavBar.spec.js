import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ADMIN, MANAGER_SR, STAFF_USER } from "../../utility/types";
import { NavBar, HamburgerNavBar } from "./NavBar";

describe("NavBar", () => {
    it("renders correctly when passed staff user props", async () => {
        const { container } = render(
            <MemoryRouter>
                <NavBar systemRole={STAFF_USER} />
            </MemoryRouter>,
        );

        const links = await screen.findAllByRole("link");
        expect(links.length).toBe(3);
        expect(screen.getByRole("button", { name: "Sign out" })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("renders correctly when passed manager props", async () => {
        const { container } = render(
            <MemoryRouter>
                <NavBar systemRole={MANAGER_SR} />
            </MemoryRouter>,
        );

        const links = await screen.findAllByRole("link");
        expect(links.length).toBe(8);
        expect(screen.getByRole("button", { name: "Sign out" })).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
    it("renders correctly when passed admin props", async () => {
        const { container } = render(
            <MemoryRouter>
                <NavBar systemRole={ADMIN} />
            </MemoryRouter>,
        );

        const links = await screen.findAllByRole("link");
        expect(links.length).toBe(8);
        expect(screen.getByRole("button", { name: "Sign out" })).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});

describe("HamburgerNavBar", () => {
    it("renders correctly when passed staff user props", async () => {
        const { container } = render(
            <MemoryRouter>
                <HamburgerNavBar systemRole={STAFF_USER} />
            </MemoryRouter>,
        );

        const links = await screen.findAllByRole("link");
        expect(links.length).toBe(3);
        expect(screen.getByRole("button", { name: "Sign out" })).toBeInTheDocument();
        expect(screen.getByRole("button", { name: "Navigation" })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("renders correctly when passed manager props", async () => {
        const { container } = render(
            <MemoryRouter>
                <HamburgerNavBar systemRole={MANAGER_SR} />
            </MemoryRouter>,
        );

        const links = await screen.findAllByRole("link");
        expect(links.length).toBe(8);
        expect(screen.getByRole("button", { name: "Sign out" })).toBeInTheDocument();
        expect(screen.getByRole("button", { name: "Navigation" })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
    it("renders correctly when passed admin props", async () => {
        const { container } = render(
            <MemoryRouter>
                <HamburgerNavBar systemRole={ADMIN} />
            </MemoryRouter>,
        );

        const links = await screen.findAllByRole("link");
        expect(links.length).toBe(8);
        expect(screen.getByRole("button", { name: "Sign out" })).toBeInTheDocument();
        expect(screen.getByRole("button", { name: "Navigation" })).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
