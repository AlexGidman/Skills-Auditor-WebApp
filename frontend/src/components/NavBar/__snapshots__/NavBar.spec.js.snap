// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerNavBar renders correctly when passed admin props 1`] = `
<div>
  <nav>
    <button
      aria-label="Navigation"
      class="button"
    >
      <img
        alt="menu"
        aria-hidden="true"
        class="hamburgerIcon"
        src="hamburger.svg"
      />
    </button>
    <ul
      class="hamburgerLinks hidden"
    >
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/mydetails"
        >
          My Details
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/myskills"
        >
          My Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/viewstaff"
        >
          View Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/addstaff"
        >
          Add Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/viewskills"
        >
          View Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/addskill"
        >
          Add Skill
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/viewcategories"
        >
          View Categories
        </a>
      </li>
      <li
        class="listItem"
      >
        <button
          class="component signoutButton"
          name="Sign Out"
          path="/login"
        >
          Sign out
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`HamburgerNavBar renders correctly when passed manager props 1`] = `
<div>
  <nav>
    <button
      aria-label="Navigation"
      class="button"
    >
      <img
        alt="menu"
        aria-hidden="true"
        class="hamburgerIcon"
        src="hamburger.svg"
      />
    </button>
    <ul
      class="hamburgerLinks hidden"
    >
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/mydetails"
        >
          My Details
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/myskills"
        >
          My Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/viewstaff"
        >
          View Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/addstaff"
        >
          Add Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/viewskills"
        >
          View Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/addskill"
        >
          Add Skill
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/viewcategories"
        >
          View Categories
        </a>
      </li>
      <li
        class="listItem"
      >
        <button
          class="component signoutButton"
          name="Sign Out"
          path="/login"
        >
          Sign out
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`HamburgerNavBar renders correctly when passed staff user props 1`] = `
<div>
  <nav>
    <button
      aria-label="Navigation"
      class="button"
    >
      <img
        alt="menu"
        aria-hidden="true"
        class="hamburgerIcon"
        src="hamburger.svg"
      />
    </button>
    <ul
      class="hamburgerLinks hidden"
    >
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/mydetails"
        >
          My Details
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component"
          href="/myskills"
        >
          My Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <button
          class="component signoutButton"
          name="Sign Out"
          path="/login"
        >
          Sign out
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`NavBar renders correctly when passed admin props 1`] = `
<div>
  <nav
    class="component"
  >
    <img
      alt="logo"
      aria-hidden="true"
      class="logo"
      src="logo.svg"
    />
    <ul>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/mydetails"
        >
          My Details
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/myskills"
        >
          My Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/viewstaff"
        >
          View Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/addstaff"
        >
          Add Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/viewskills"
        >
          View Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/addskill"
        >
          Add Skill
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/viewcategories"
        >
          View Categories
        </a>
      </li>
      <li
        class="listItem"
      >
        <button
          class="component signoutButton"
          name="Sign Out"
          path="/login"
        >
          Sign out
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`NavBar renders correctly when passed manager props 1`] = `
<div>
  <nav
    class="component"
  >
    <img
      alt="logo"
      aria-hidden="true"
      class="logo"
      src="logo.svg"
    />
    <ul>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/mydetails"
        >
          My Details
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/myskills"
        >
          My Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/viewstaff"
        >
          View Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/addstaff"
        >
          Add Staff
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/viewskills"
        >
          View Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/addskill"
        >
          Add Skill
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/viewcategories"
        >
          View Categories
        </a>
      </li>
      <li
        class="listItem"
      >
        <button
          class="component signoutButton"
          name="Sign Out"
          path="/login"
        >
          Sign out
        </button>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`NavBar renders correctly when passed staff user props 1`] = `
<div>
  <nav
    class="component"
  >
    <img
      alt="logo"
      aria-hidden="true"
      class="logo"
      src="logo.svg"
    />
    <ul>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/mydetails"
        >
          My Details
        </a>
      </li>
      <li
        class="listItem"
      >
        <a
          class="component link"
          href="/myskills"
        >
          My Skills
        </a>
      </li>
      <li
        class="listItem"
      >
        <button
          class="component signoutButton"
          name="Sign Out"
          path="/login"
        >
          Sign out
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
